src/Mvc/Mvc.ViewFeatures/src/Builder/ControllerActionEndpointConventionBuilderResourceCollectionExtensions.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Endpoints;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.AspNetCore.Builder;
+
+/// <summary>
+/// Extensions for <see cref="ControllerActionEndpointConventionBuilder"/>.
+/// </summary>
+public static class ControllerActionEndpointConventionBuilderResourceCollectionExtensions
+{
+    /// <summary>
+    /// Adds a <see cref="ResourceAssetCollection"/> metadata instance to the endpoints.
+    /// </summary>
+    /// <param name="builder">The <see cref="ControllerActionEndpointConventionBuilder"/>.</param>
+    /// <param name="manifestPath">The manifest associated with the assets.</param>
+    /// <returns></returns>
+    public static ControllerActionEndpointConventionBuilder WithStaticAssets(
+        this ControllerActionEndpointConventionBuilder builder,
+        string manifestPath = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        if (builder.Items.TryGetValue("__EndpointRouteBuilder", out var endpointBuilder))
+        {
+            var (resolver, registered) = builder.Items.TryGetValue("__ResourceCollectionResolver", out var value)
```

Comment:
Nit: More repeated strings (not in the same file, but across projects).

Maybe this is something already on your TODO list, in which case I apologize if it's not helpful to comment on these!